### PR TITLE
Remove awkwardness of suit pockets

### PR DESCRIPTION
**Pull request by XElectricX**
## About The Pull Request

Originally this was just going to set the storage name to the armor name but my brain 2 cells continued rubbing. Made a new proc to handle displaying the message so there wouldn't be the headache of maintaining a large copy-pasted proc just so it could have a different message. Pockets are also now their own thing instead of just being a child of the general storage object. Found it weird that it was tied to it since internal suit pockets should not be handled as if they were your normal storage module. They can be balanced separately and such.

## Why It's Good For The Game

It bothered me how if you put an item into your normal looking coat it would say you put it into a "General Purpose Storage module" instead. Less jank is good.

## Changelog
:cl:
code: Suit pockets no longer appear as if they are just bolted-on storage modules. Are now their own independent object.
/:cl:

### CODE REVIEW

File: code/modules/clothing/modular_armor/attachments/storage.dm [hunk 1]
```diff
@@ -91,6 +91,33 @@
 		/obj/item/stack,
 	)
 
+/* Pockets */
+/obj/item/armor_module/storage/pocket
+	icon_state = ""
+	item_state = ""
```
[the remark this answers is not in the document]
Comment (XElectricX) [reply code/modules/clothing/modular_armor/attachments/storage.dm:97]: One does not remove pockets.

[the remark this answers is not in the document]
Comment (XElectricX) [reply code/modules/clothing/modular_armor/attachments/storage.dm:97]: The flag is to make it unremovable? How misleading.

[the remark this answers is not in the document]
Comment (TiviPlus) [reply code/modules/clothing/modular_armor/attachments/storage.dm:97]: no adding that code removes the ATTACH_REMOVABLE flag

[the remark this answers is not in the document]
Comment (TiviPlus) [reply code/modules/clothing/modular_armor/attachments/storage.dm:97]: though it should probably be ATTACH_UNREMOVABLE instead tbh

**Pull request merged**